apps/solarmanagerch/solar_manager_ch.star

```diff
@@ -249,75 +251,72 @@ def main(config):
-            data["production"] = sum_data["data"][0]["production"]
-
-            data["autarky_day"] = get_autarky_percent(site_id, api_key, tz, "day")
-            data["autarky_24h"] = get_autarky_percent(site_id, api_key, tz, "24h")
-            data["autarky_month"] = get_autarky_percent(site_id, api_key, tz, "month")
-            data["autarky_year"] = get_autarky_percent(site_id, api_key, tz, "year")
-
-            data["aux_sensor_day"] = None
-            data["aux_sensor_week"] = None
-            data["aux_sensor_month"] = None
-
-            if config.get("aux_sensor_id", "") != "":
-                data["aux_sensor_day"] = get_aux_sensor_data(config.get("aux_sensor_id"), api_key, "day")
-                data["aux_sensor_week"] = get_aux_sensor_data(config.get("aux_sensor_id"), api_key, "week")
-                data["aux_sensor_month"] = get_aux_sensor_data(config.get("aux_sensor_id"), api_key, "month")
-
-            # We store custom data and data from multiple http calls so http cache won't suffice here.
-            cache.set(site_id, json.encode(data), ttl_seconds = CACHE_TTL)
-        else:
-            print("using cache")
-            data = json.decode(data)  # data from cache is json so need to decode.
-            if "has_battery" in data and data["has_battery"] == True:
-                has_battery = True
-            if config.get("aux_sensor_id", "") != "" and data.get("aux_sensor_day", "") == "":
-                print("fetching aux_sensor data outside of cache")
-                data["aux_sensor_day"] = get_aux_sensor_data(config.get("aux_sensor_id"), api_key, "day")
-                data["aux_sensor_week"] = get_aux_sensor_data(config.get("aux_sensor_id"), api_key, "week")
-                data["aux_sensor_month"] = get_aux_sensor_data(config.get("aux_sensor_id"), api_key, "month")
+        url = URL_CUR.format(site_id)
+        data = dict()
+        rep = http.get(
+            url,
+            headers = {
+                "Accept": "application/json",
+                "Authorization": "Basic " + api_key,
+            },
+            ttl_seconds = 300,  # 5 minutes
+        )
+        if rep.status_code != 200:
+            print(rep.body())
+            return render_fail(rep)
+
+        cur_data = json.decode(rep.body())
+        data["currentPowerConsumption"] = float(cur_data["currentPowerConsumption"])
+        data["currentPvGeneration"] = float(cur_data["currentPvGeneration"])
+        data["currentBatteryChargeDischarge"] = 0.0  # set to zero just to be safe
+        data["soc"] = 0  # set to zero just to be safe
+
+        if "currentBatteryChargeDischarge" in cur_data and "soc" in cur_data:
+            data["currentBatteryChargeDischarge"] = float(cur_data["currentBatteryChargeDischarge"])
+            data["soc"] = cur_data["soc"]
+            data["has_battery"] = True
+            has_battery = True
+
+        url = URL_SUM.format(site_id)
+        rep = http.get(
```

Comment:
No `ttl_seconds` on this call?

Reply:
oh wow, can't believe I missed that. updating.